src/main/java/org/elasticsearch/index/engine/internal/InternalEngine.java

```diff
@@ -1135,37 +1150,32 @@ public int compare(Segment o1, Segment o2) {
 
     @Override
     public void close() throws ElasticsearchException {
-        logger.debug("close now acquire writeLock");
-        try (InternalLock _ = writeLock.acquire()) {
```

Comment:
I'm worried that no acquiring the write lock will cause in flight indexing request to fail with a IndexFailedEngineException , even if the close was due to a lower level error.  Also I like the semantics of letting all in flight operations to complete before closing (due to normal operations)